src/backend/tests/unit/components/tools/test_file_manipulation.py

```
@@ -0,0 +1,502 @@
+        """
+        Tests inserting a string at a specified line in a file using the 'insert_at_line' tool.
+        
+        Creates a file, inserts a new string at the given line, and verifies that the insertion
+        was successful and the file content was updated accordingly.
+        """
+        file_path = tmp_path / "insert.txt"
+        await anyio.Path(file_path).write_text("line1\nline2\n")
+        component = FileManipulation(workspace_folder=str(tmp_path))
+        tools = component.build_toolkit()
+        insert_at_line_tool: Tool = next(t for t in tools if t.name == "insert_at_line")
+        result = await insert_at_line_tool.ainvoke({"path": "insert.txt", "insert_line": 2, "new_str": "inserted"})
```

Pylint (3.3.7)

[convention] Line too long (115/100)

(C0301)